app/views/people_skills/index.html.haml

```diff
@@ -16,7 +16,14 @@
   %turbo-frame{id: "search-results"}
     %div.border.p-3.rounded-bottom.search-results
       - if entries.size == 0
-        = ti "search.no_results"
+        - if filter_params.skill_ids.empty? || filter_params.skill_ids.any? { |x| x.to_i <= 0 }
+          = ti "search.no_skill"
+        - else
```

Comment:
This is all quite overcomplicated and not readable at all.
How about building the whole string in the controller?
So basically all you would have in the view is:
`= search_result_message(filter_params)`

Then in the controller add the above method and build the string dynamically. This allows you to clean up quite a bit of code and split it up into several methods (which you should).